src/relation.js

```diff
@@ -296,7 +296,8 @@ export default RelationBase.extend({
         return this.isInverse() && this.isThrough() ? m.pivot.id :
           m.pivot.get(this.key('foreignKey'));
       } else {
-        return this.isInverse() ? m.id : m.get(this.key('foreignKey'));
+        return this.isInverse() ? m.get(this.targetIdAttribute) :
+          m.get(this.key('foreignKey'));
```

Comment:
@ricardogama, if done how I [commented](https://github.com/tgriesser/bookshelf/pull/1397#issuecomment-250721474), you would need this change.